src/test/java/com/endava/internship/collections/TestStudentList.java

```diff
@@ -0,0 +1,483 @@
+package com.endava.internship.collections;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.stream.Stream;
+
+class TestStudentList {
+    static StudentList students_List;
+    static StudentList emptyList;
+    static Student student1 = new Student("Bob1", LocalDate.parse("2020-01-01"), "Good student");
+    static Student student2 = new Student("Bob2", LocalDate.parse("2020-01-02"), "Good student");
+    static Student student3 = new Student("Bob3", LocalDate.parse("2020-01-03"), "Good student");
+    static Student student4 = new Student("Bob4", LocalDate.parse("2020-01-04"), "Good student");
+    static Student student5 = new Student("Bob5", LocalDate.parse("2020-01-05"), "Good student");
+    static Student student6 = new Student("Bob6", LocalDate.parse("2020-01-06"), "Good student");
+    static Student student7 = new Student("Bob7", LocalDate.parse("2020-01-07"), "Good student");
+    static Student student8 = new Student("Bob8", LocalDate.parse("2020-01-08"), "Good student");
+    static Student student9 = new Student("Bob9", LocalDate.parse("2020-01-09"), "Good student");
+    static Student student10 = new Student("Bob10", LocalDate.parse("2020-01-10"), "Good student");
+    static Student student11 = new Student("Bob11", LocalDate.parse("2020-01-11"), "Good student");
+    static Student student12 = new Student("Bob12", LocalDate.parse("2020-01-12"), "Good student");
+    static Student student0 = new Student("Jery", LocalDate.parse("2020-01-19"), "Best Student"); // Note this objects will not be added in setUp()
+
+    private static Stream<Arguments> studentsStream() {
+        return Stream.of(
+                arguments(student1),
+                arguments(student2),
+                arguments(student3),
+                arguments(student4),
+                arguments(student5),
+                arguments(student6),
+                arguments(student7),
+                arguments(student8),
+                arguments(student9),
+                arguments(student10),
+                arguments(student11),
+                arguments(student12));
+    }
+
+    private void createThreads(int numberOfThreads, StudentList targetList, String action) {
+        for (int i = 0; i < numberOfThreads; i++) {
+            new Thread(threadProperties(targetList, action)).start();
+        }
+    }
+
+    private static Runnable threadProperties(StudentList targetList, String action) {
```

Comment:
it wasn't in task scope to test concurrency of the collection, so ➕  for the initiative 👍🏻 😄